packages/web/src/store/account/sagas.ts

```diff
@@ -0,0 +1,198 @@
+      yield call(audiusBackendInstance.updateBrowserNotifications, {
+        subscription: pushManagerSubscription,
+        enabled: true
+      })
+      yield* put(setBrowserNotificationSettingsOn())
+    } else if (
+      window.Notification &&
+      window.Notification.permission !== Permission.DENIED
+    ) {
+      const subscription = yield* call(subscribePushManagerBrowser)
+      const enabled = !!subscription
+      if (enabled) {
+        yield* put(setBrowserNotificationPermission(Permission.GRANTED))
+        yield* put(setBrowserNotificationEnabled(true, false))
+        yield* call(audiusBackendInstance.updateBrowserNotifications, {
+          subscription,
+          enabled: true
+        })
+      } else {
+        yield* put(setBrowserNotificationPermission(Permission.DENIED))
+      }
+    }
+  }
+  // Note: you cannot request safari permission from saga
+  // it must be initiated from a user action (in the component)
+  if (isSafariPushAvailable) {
+    const safariSubscription = yield* call(getSafariPushBrowser)
+    if (safariSubscription.permission === Permission.GRANTED) {
+      yield* call(
+        audiusBackendInstance.registerDeviceToken,
+        safariSubscription.deviceToken,
+        'safari'
+      )
+      yield* put(setBrowserNotificationEnabled(true, false))
+      yield* put(setBrowserNotificationSettingsOn())
+    }
+  }
+}
+
+export function* unsubscribeBrowserPushNotifcations() {
+  removeHasRequestedBrowserPermission()
+  const browserPushSubscriptionStatus = yield* call(
+    fetchBrowserPushNotifcationStatus
+  )
+  if (
+    browserPushSubscriptionStatus === Permission.GRANTED &&
+    isPushManagerAvailable
+  ) {
+    const subscription = yield* call(getPushManagerBrowserSubscription)
+    yield* call(audiusBackendInstance.disableBrowserNotifications, {
+      subscription
+    })
+  } else if (
+    browserPushSubscriptionStatus === Permission.GRANTED &&
+    isSafariPushAvailable
+  ) {
+    const safariSubscription = yield* call(getSafariPushBrowser)
+    if (safariSubscription.permission === Permission.GRANTED) {
+      yield* call(
+        audiusBackendInstance.deregisterDeviceToken,
+        safariSubscription.deviceToken
+      )
+    }
+  }
+}
+
+function* getBrowserPushNotifcations() {
+  yield* takeEvery(
+    accountActions.fetchBrowserPushNotifications.type,
+    fetchBrowserPushNotifcationStatus
+  )
+}
+
+function* watchShowPushNotificationConfirmation() {
+  yield* takeEvery(
+    accountActions.showPushNotificationConfirmation.type,
+    showPushNotificationConfirmation
+  )
+}
+
+function* subscribeBrowserPushNotification() {
+  yield* takeEvery(
+    accountActions.subscribeBrowserPushNotifications.type,
+    subscribeBrowserPushNotifcations
+  )
+}
+
+function* unsubscribeBrowserPushNotification() {
+  yield* takeEvery(
+    accountActions.unsubscribeBrowserPushNotifications.type,
+    unsubscribeBrowserPushNotifcations
+  )
+}
+
+export default function sagas() {
```

Comment:
awesome work untangling all this

Reply:
> Looks great! so for the mobile account stuff, will that just use common, or also some mobile specific stuff on the reloaded branch only? or a guess rn the common account stuff does the message passing, which will be resolved like mentioned?

Yeah the mobile aagas are still doing message passing. In the next PR I will get rid of that and see if mobile specific sagas are required